blogposts/more-intuitive-code-search-on-sourcegraph-dot-com.md

```diff
@@ -0,0 +1,59 @@
+published: true
+---
+
+<p style="text-align: center">
+ <img src="/blog/sourcegraph-com-usage.png" />
+ <br/><small>Sourcegraph.com stats as of August 2019</small>
+</p>
+
+[Sourcegraph.com](https://sourcegraph.com) can now search across the top 10,000 open source repositories on GitHub so you can find usage examples of popular projects.
+
+## Using Sourcegraph.com like a regular search engine
+
+[Sourcegraph.com](https://sourcegraph.com) has always allowed developers to browse any open source repository from GitHub, but it wasn't possible to perform a search without adding a `repo` filter to limit the number of repositories that needed to be searched (private Sourcegraph instances don't have this limitation). This meant [Sourcegraph.com](https://sourcegraph.com) was missing a crucial use case—finding usage examples for a library, class, method, or function, from a set or reputable, and predefined open source repositories.
+
+Let's say you've recently heard about Google's [Distroless container image](https://github.com/GoogleContainerTools/distroless), and are curious about its usage. Previously, without specifying any repositories, your search would've returned this:
+
+![](/blog/too-many-matching-repos.png)
+
+After some [Sourcegraph.com](https://sourcegraph.com) specific code changes ([#4959](https://github.com/sourcegraph/sourcegraph/pull/4959/files) and [#5062](https://github.com/sourcegraph/sourcegraph/pull/5062/files)), Sourcegraph (by default), will now search the top 10,000 open source repositories from GitHub (ordered by watch count) if a [`repo` filter](https://docs.sourcegraph.com/user/search/queries) was not provided. This means a query for a [distroless based Dockerfile](https://sourcegraph.com/search?q=file:Dockerfile+FROM+gcr.io/distroless) now returns a useful set of results to explore.
+
+![](/blog/distroless-image-results.png)
+
+## Sourcegraph.com as a research and usage discovery tool
+
+Let's explore some research and usage possibilities, such as examples of [Python's Pipenv within Dockerfiles](https://sourcegraph.com/search?q=pipenv+f:Dockerfile):
+
+![](/blog/pipenv-search.png)
+
+Or how developers are using [React in Typescript](https://sourcegraph.com/search?q=react%5C.+lang:typescript):
+
+![](/blog/react-typescript-search.png)
+
+We're excited by the increased discoverability of code on [Sourcegraph.com](https://sourcegraph.com) for languages, packages, libraries, and frameworks. There is also the added benefit of [Sourcegraph.com](https://sourcegraph.com) working for a first time user (when they don't necessarily know the [`repo` filter](https://docs.sourcegraph.com/user/search/queries) exists).
+
+Here are some additional searches:
+
+ - [Dockerfile's for Python Flask apps using uwsgi](https://sourcegraph.com/search?q=file:Dockerfile+%5EFROM%5Cs%2Bpython%5C:+uwsgi+flask+count:25)
+ - [TypeScript examples using `tap` from RxJS](https://sourcegraph.com/search?q=lang:typescript+rxjs+tap%5C%28+count:25)
```

Comment:
I had trouble finding this like this, but got results with `"tap(val =>" index:only`.

Reply:
None of these links appear to actually work right now. Is indexing disabled?

![image](https://user-images.githubusercontent.com/754768/63306825-163b0280-c2a1-11e9-9af5-7fde5d6e3b29.png)

Reply:
That's been fixed, thanks!

Reply:
I turned it off by setting maxReposToSearch=50 after chatting with Keegan a bit about the poor performance we observed today on several queries. Apparently it's been falling back on searcher for some repos even though I see over 10k index files in /data on the indexer. Running the queries with `index:only` gives fast results, so I'm working on making that the default, as [suggested by Keegan](https://sourcegraph.slack.com/archives/CHEKCRWKV/p1566250606138800?thread_ts=1566248349.136900&cid=CHEKCRWKV). Here is the issue: https://github.com/sourcegraph/sourcegraph/issues/5301.